client/my-sites/site-settings/date-time-format/date-time-format-options.jsx

```diff
@@ -0,0 +1,338 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import { includes, startsWith } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Card from 'components/card';
+import ExternalLink from 'components/external-link';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormInput from 'components/forms/form-text-input';
+import FormLabel from 'components/forms/form-label';
+import FormRadio from 'components/forms/form-radio';
+import FormSelect from 'components/forms/form-select';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import SectionHeader from 'components/section-header';
+import { phpToMomentDatetimeFormat } from 'lib/formatting';
+import wrapSettingsForm from '../wrap-settings-form';
+
+/**
+ * Module constants
+ */
+const defaultDateFormats = [ 'F j, Y', 'Y-m-d', 'm/d/Y', 'd/m/Y' ];
+const defaultTimeFormats = [ 'g:i a', 'g:i A', 'H:i' ];
+
+export class DateTimeFormatOptions extends Component {
+	state = {
+		customDateFormat: false,
+		customTimeFormat: false,
+		isLoadingSettings: true,
+	};
+
+	isDefaultFormat = ( defaultFormats, format ) => ! includes( defaultFormats, format );
+
+	handleRadio = event => {
+		const { name: field, value: format } = event.currentTarget;
+		this.props.handleRadio( event );
+		if ( 'date_format' === field ) {
+			this.setState( {
+				customDateFormat: this.isDefaultFormat( defaultDateFormats, format ),
+			} );
+		} else if ( 'time_format' === field ) {
+			this.setState( {
+				customTimeFormat: this.isDefaultFormat( defaultTimeFormats, format ),
+			} );
+		}
+	}
```

Comment:
even though the _UI_ is similar the _intentions_ are different here. I might favor seeing two separate functions which each handle their own data than one which intermingles data types (and avoids the `handle___` naming)

```js
chooseDateFormat = event => {
	const { value: format } = event.currentTarget;

	this.setState( {
		customDateFormat: ! includes( defaultDateFormats, format ),
	}, () => this.props.handleRadio( event ) );
}

chooseTimeFormat = event => {
	const { value: format } = event.currentTarget;

	this.setState( {
		customTimeFormat: ! includes( defaultTimeFormats, format ),
	}, () => this.props.handleRadio( event ) );
}
```

or further abstracted…

```js
chooseFormat = ( name, defaults ) => event => {
	const { value: format } = event.currentTarget;

	this.setState( {
		[ `custom${ capitalize( name ) }Format` ]: ! includes( defaults, format ),
	}, () => this.props.handleRadio( event ) );
}

chooseDateFormat = this.chooseFormat( 'date', defaultDateFormats );

chooseTimeFormat = this.chooseFormat( 'time', defaultTimeFormats );
```